### DISCUSSION TITLE
Linux->Macos cross-compilation error

### DISCUSSION THREAD

**Discussion by cstkingkey** [fltk-rs/fltk-rs#1565] [General]
https://github.com/fltk/fltk/commit/377ff11ec9995d2842257daa7edde4e57320512e
fltk use new method to link frameworks, can ftlk-rs adapt to it?

**Comment by MoAlyousef** [comment]
I'll need more info.
What's the build error? What version of macos?
AFAIK Rust doesn't support a weak_framework linker argument in the build.rs file. So if something is required it will have to be passed to the linker as far as Rust is concerned.

**Comment by cstkingkey** [comment]
Undefined symbols for architecture x86_64:
            "_OBJC_CLASS_$_UTType", referenced from:
                objc-class-ref in libfltk_sys-df8b95a7da06d2e3.rlib(Fl_Native_File_Chooser_MAC.mm.o)
            "___isPlatformVersionAtLeast", referenced from:
                -[FLsaveDelegate control_allowed_types:] in libfltk_sys-df8b95a7da06d2e3.rlib(Fl_Native_File_Chooser_MAC.mm.o)
          ld: symbol(s) not found for architecture x86_64
https://github.com/fltk-rs/fltk-rs/discussions/1541

**Comment by cstkingkey** [comment]
workaround it by setting CARGO_ENCODED_RUSTFLAGS="-Clink-args=-weak_framework UniformTypeIdentifiers  -lSystem /usr/lib/llvm-17/lib/clang/17/lib/darwin/libclang_rt.osx.a"

**Reply by MoAlyousef** [reply]
> workaround it by setting CARGO_ENCODED_RUSTFLAGS="-Clink-args=-weak_framework UniformTypeIdentifiers -lSystem /usr/lib/llvm-17/lib/clang/17/lib/darwin/libclang_rt.osx.a"

Thank you for posting the workaround. Of note this is required when cross-compiling from Linux to Macos. 
I think once Cargo supports weak_framework it shouldn’t be needed
